### PR TITLE
[D1] Add Time Travel documentation

### PR DESCRIPTION
**Pull request by elithrar**
This PR:

- [x] Adds new documentation for Time Travel
- [x] Updates the wrangler docs to detail the `wrangler d1 time-travel` commands: https://developers.cloudflare.com/workers/wrangler/commands/#d1
- [x] Updates the existing Backups documentation to reference Time Travel for new databases
- [x] Adds a /d1/changelog entry for Time Travel

Relates to https://github.com/cloudflare/workers-sdk/pull/3579

### CODE REVIEW

File: content/d1/changelog.md [hunk 1]
```diff
@@ -7,6 +7,16 @@ rss: file
 
 # Changelog
 
+## 2023-07-14
+
+### Time Travel
+
+[Time Travel](/d1/learning/time-travel/) is now available. Time Travel allows you to restore a D1 database back to any minute within the last 30 days (Workers paid plans) or 7 days (Workers free plans), at no additional cost for storage or restore operations.
```
Comment (Maddy-Cloudflare) [inline content/d1/changelog.md]:
```suggestion
[Time Travel](/d1/learning/time-travel/) is now available. Time Travel allows you to restore a D1 database back to any minute within the last 30 days (Workers Paid plan) or 7 days (Workers Free plan), at no additional cost for storage or restore operations.
```

File: content/d1/learning/time-travel.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+title: Time Travel & Backups
+pcx_content_type: concept
+weight: 3
+---
+
+# Time Travel
+
+Time Travel is D1's approach to backups and point-in-time-recovery, and allows you to restore a database to any minute within the last 30 days.
+
+* You do not need to enable Time Travel: it is always on.
```
Comment (Maddy-Cloudflare) [inline content/d1/learning/time-travel.md]:
```suggestion
* You do not need to enable Time Travel. It is always on.
```

File: content/d1/learning/time-travel.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+* Database history and restoring a database incur no additional costs.
+* Time Travel automatically creates [bookmarks](#terminology) on your behalf: you do not need to manually trigger or remember to initiate a backup.
```
Comment (Maddy-Cloudflare) [inline content/d1/learning/time-travel.md]:
```suggestion
* Time Travel automatically creates [bookmarks](#terminology) on your behalf. You do not need to manually trigger or remember to initiate a backup.
```

File: content/d1/learning/time-travel.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+Databases using D1's [new storage subsystem](https://blog.cloudflare.com/d1-turning-it-up-to-11/) can use Time Travel. Time Travel replaces the [snapshot-based backups](/d1/learning/backups/) used for legacy alpha databases.
+
+To understand which storage subsystem your database uses, run `wrangler d1 info YOUR_DATABASE` and inspect the `version` field in the output. Databases with `version: beta` support the new Time Travel API. Databases with `version: alpha` only support the older, snapshot-based backup API.
+
+{{</Aside>}}
+
+## Terminology
+
+Time Travel introduces the concept of a "bookmark" to D1: a bookmark represents the state of a database at a specific point in time, and is effectively an append-only log.
```
Comment (Maddy-Cloudflare) [inline content/d1/learning/time-travel.md]:
```suggestion
Time Travel introduces the concept of a "bookmark" to D1. A bookmark represents the state of a database at a specific point in time, and is effectively an append-only log.
```

File: content/d1/learning/time-travel.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
+---
+title: Time Travel & Backups
```
Comment (Maddy-Cloudflare) [inline content/d1/learning/time-travel.md]:
```suggestion
title: Time Travel and backups
```

File: content/d1/learning/time-travel.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+```sh
+$ wrangler d1 time-travel restore YOUR_DATABASE --at-timestamp=UNIX_TIMESTAMP
+
+# Example output:
+...
+```
+
+Note that:
+
+* Queries in flight will be cancelled, and an error returned to the client
```
Comment (Maddy-Cloudflare) [inline content/d1/learning/time-travel.md]:
```suggestion
* Queries in flight will be cancelled, and an error returned to the client.
```

File: content/d1/learning/time-travel.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+* The restore operation will return a [bookmark](#terminology) that allows you to [undo](#undo-a-restore) and revert the database.
+
+## Undo a restore
+
+You can undo a restore by:
+
+* Taking note of the previous bookmark returned as part of a `wrangler d1 restore` operation
```
Comment (Maddy-Cloudflare) [inline content/d1/learning/time-travel.md]:
```suggestion
* Taking note of the previous bookmark returned as part of a `wrangler d1 restore` operation.
```

File: content/d1/learning/time-travel.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+* Restore directly to a bookmark in the past, prior to your last restore.
+
+```sh
+# Get a historical bookmark
+$ wrangler d1 time-travel info 
+```
+
+## Notes
+
+* You can quickly get the Unix timestamp from the command-line in macOS and Windows via `date %+s`
```
Comment (Maddy-Cloudflare) [inline content/d1/learning/time-travel.md]:
```suggestion
* You can quickly get the Unix timestamp from the command-line in macOS and Windows via `date %+s`.
```

File: content/d1/learning/time-travel.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+To retrieve the bookmark for a timestamp in the past, pass the `--at-timestamp` flag with a valid Unix or RFC3339 timestamp:
+
+```sh
+# Using an RFC3339 timestamp, including the timezone:
+$ wrangler d1 time-travel info YOUR_DATABASE --timestamp='2023-07-09T17:31:11+00:00"
+```
+
+## Restore a database
+
+To restore a database a specific point-in-time:
```
Comment (Maddy-Cloudflare) [inline content/d1/learning/time-travel.md]:
```suggestion
To restore a database to a specific point-in-time:
```

File: content/d1/learning/time-travel.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+* Queries in flight will be cancelled, and an error returned to the client
+* The restore operation will return a [bookmark](#terminology) that allows you to [undo](#undo-a-restore) and revert the database.
+
+## Undo a restore
+
+You can undo a restore by:
+
+* Taking note of the previous bookmark returned as part of a `wrangler d1 restore` operation
+* Restore directly to a bookmark in the past, prior to your last restore.
```
Comment (Maddy-Cloudflare) [inline content/d1/learning/time-travel.md]:
```suggestion
* Restoring directly to a bookmark in the past, prior to your last restore.
```

File: content/d1/changelog.md [hunk 11]
```diff
@@ -7,6 +7,16 @@ rss: file
...
+
+Refer to the [Time Travel documentation](/d1/learning/time-travel/) to learn how to travel backwards in time.
```
Comment (Maddy-Cloudflare) [inline content/d1/changelog.md]:
```suggestion
Refer to the [Time Travel](/d1/learning/time-travel/) documentation to learn how to travel backwards in time.
```

File: content/d1/learning/time-travel.md [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+
+To understand which storage subsystem your database uses, run `wrangler d1 info YOUR_DATABASE` and inspect the `version` field in the output. Databases with `version: beta` support the new Time Travel API. Databases with `version: alpha` only support the older, snapshot-based backup API.
+
+{{</Aside>}}
+
+## Terminology
+
+Time Travel introduces the concept of a "bookmark" to D1: a bookmark represents the state of a database at a specific point in time, and is effectively an append-only log.
+
+* Bookmarks are lexicographically sortable: sorting a list of bookmarks will order them from oldest-to-newest
```
Comment (Maddy-Cloudflare) [inline content/d1/learning/time-travel.md]:
```suggestion
* Bookmarks are lexicographically sortable. Sorting orders a list of bookmarks from oldest-to-newest.
```

**Review by Maddy-Cloudflare — CHANGES_REQUESTED** [review]
Will need to review this again once it's finished.

**Comment by elithrar** [comment]
@Maddy-Cloudflare - ready for final review!

File: content/workers/wrangler/commands.md [hunk 13]
```diff
@@ -181,6 +181,40 @@ $ wrangler d1 execute <DATABASE_NAME> [OPTIONS]
...
+```
+
+{{<definitions>}}
+
+- `DATABASE_NAME` {{<type>}}string{{</type>}} {{<prop-meta>}}required{{</prop-meta>}}
+  - The name of the D1 database to execute a query on.
+- `--bookmark` {{<type>}}string{{</type>}}
+  - ...
+- `--timestamp` {{<type>}}string{{</type>}}
+  - ...
```
Comment (rozenmd) [inline content/workers/wrangler/commands.md]: these `...` should probably have more info, but can iterate on it

**Review by rozenmd — APPROVED** [review]
LGTM apart from the `...`s in `workers/wrangler/commands.md`, but can fix 'em later

**Review by Maddy-Cloudflare — APPROVED** [review]

File: content/workers/wrangler/commands.md (hunk 13 above)
Comment (Maddy-Cloudflare) [reply content/workers/wrangler/commands.md]: @elithrar are you planning on adding more information here?

**Review by Maddy-Cloudflare — CHANGES_REQUESTED** [review]

File: content/workers/wrangler/commands.md (hunk 13 above)
Comment (elithrar) [reply content/workers/wrangler/commands.md]: Fixed both.

File: content/workers/wrangler/commands.md [hunk 14]
```diff
@@ -181,6 +181,40 @@ $ wrangler d1 execute <DATABASE_NAME> [OPTIONS]
...
+```sh
+$ wrangler d1 time-travel info <DATABASE_NAME> [OPTIONS]
+```
+
+{{<definitions>}}
+
+- `DATABASE_NAME` {{<type>}}string{{</type>}} {{<prop-meta>}}required{{</prop-meta>}}
+  - The name of the D1 database to execute a query on.
+- `--timestamp` {{<type>}}string{{</type>}}
+  - A Unix timestamp or JavaScript date-time string within the last 30 days.
```
Comment (Maddy-Cloudflare) [inline content/workers/wrangler/commands.md]:
```suggestion
  - A Unix timestamp or JavaScript date-time `string` within the last 30 days.
```

File: content/workers/wrangler/commands.md [hunk 15]
```diff
@@ -181,6 +181,40 @@ $ wrangler d1 execute <DATABASE_NAME> [OPTIONS]
...
+```
+
+{{<definitions>}}
+
+- `DATABASE_NAME` {{<type>}}string{{</type>}} {{<prop-meta>}}required{{</prop-meta>}}
+  - The name of the D1 database to execute a query on.
+- `--bookmark` {{<type>}}string{{</type>}}
+  - A D1 bookmark representing the state of a database at a specific point in time.
+- `--timestamp` {{<type>}}string{{</type>}}
+  - A Unix timestamp or JavaScript date-time string within the last 30 days.
```
Comment (Maddy-Cloudflare) [inline content/workers/wrangler/commands.md]:
```suggestion
  - A Unix timestamp or JavaScript date-time `string` within the last 30 days.
```

File: content/d1/learning/time-travel.md [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+⚠️ The current bookmark is '00000085-0000024c-00004c6d-8e61117bf38d7adb71b934ebbf891683'
+⚡️ To restore to this specific bookmark, run:
+ `wrangler d1 time-travel restore YOUR_DATABASE --bookmark=00000085-0000024c-00004c6d-8e61117bf38d7adb71b934ebbf891683`
+```
+
+## Notes
+
+* You can quickly get the Unix timestamp from the command-line in macOS and Windows via `date %+s`.
+* Time Travel does not yet allow you to clone or fork an existing database to a new copy. In the future, Time Travel will allow you to easily fork (clone) an existing database into a new database, or overwrite an existing database.
+* You can restore a database back to a point in time up to 30 days in the past (Workers Paid plans) or 7 days (Workers free plans). See the [Limits](/d1/platform/limits/) page for details on Time Travel's limits.
```
Comment (Maddy-Cloudflare) [inline content/d1/learning/time-travel.md]:
```suggestion
* You can restore a database back to a point in time up to 30 days in the past (Workers Paid plan) or 7 days (Workers Free plan). See the [Limits](/d1/platform/limits/) page for details on Time Travel's limits.
```

File: content/d1/learning/time-travel.md [hunk 17]
```diff
@@ -0,0 +1,119 @@
...
+* The restore operation will return a [bookmark](#terminology) that allows you to [undo](#undo-a-restore) and revert the database.
+
+## Undo a restore
+
+You can undo a restore by:
+
+* Taking note of the previous bookmark returned as part of a `wrangler d1 time-travel restore` operation
+* Restoring directly to a bookmark in the past, prior to your last restore.
+
+To fetch a bookmark from an earlier state
```
Comment (Maddy-Cloudflare) [inline content/d1/learning/time-travel.md]:
```suggestion
To fetch a bookmark from an earlier state:
```

**Review by Maddy-Cloudflare — APPROVED** [review]

**Pull request merged**